optaplanner-docs/src/main/asciidoc/Chapter-Optimization_algorithms/Chapter-Optimization_algorithms.adoc

```diff
@@ -35,14 +35,13 @@ image::Chapter-Optimization_algorithms/searchSpaceSizeCalculation.png[align="cen
 This search space size calculation includes infeasible solutions (if they can be represented by the model), because:
 
 * The optimal solution might be infeasible.
-* There are many types of hard constraints which cannot be incorporated in the formula practically. For example in Cloud Balancing, try incorporating the CPU capacity constraint in the formula.
+* There are many types of hard constraints that cannot be incorporated in the formula practically. For example, in Cloud Balancing, try incorporating the CPU capacity constraint in the formula.
 
-Even in cases were adding some of the hard constraints in the formula is practical, for example Course Scheduling, the resulting search space is still huge.
+Even in cases where adding some of the hard constraints in the formula is practical (for example, Course Scheduling), the resulting search space is still huge.
 ====
 
-An algorithm that checks every possible solution (even with pruning such as in <<branchAndBound,Branch And Bound>>) can easily run for billions of years on a single real-life planning problem.
-What we really want is to **find the best solution in the limited time at our
-    disposal**.
+An algorithm that checks every possible solution (even with pruning, such as in <<branchAndBound,Branch And Bound>>) can easily run for billions of years on a single real-life planning problem.
+The goal is to find the best solution in the available timeframe.
```

Comment:
Maybe "The goal" -> "The mission" or "The challenge"? the problem with "goal" is it might be confusing with "goal programming" which is very similar to "constraint programming".